// serverless-plugin-typescript/dist/tests/typescript.pluginSkipNonNode.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const TypeScriptPlugin = require("../src/index");
describe('TypeScriptPlugin', () => {
    it('rootFileNames includes only node runtimes', () => {
        const slsInstance = {
            cli: {
                log: jest.fn()
            },
            config: {
                servicePath: 'servicePath'
            },
            service: {
                provider: {
                    name: 'aws',
                    runtime: 'nodejs99'
                },
                functions: {
                    func1: {
                        handler: 'java-fn',
                        runtime: 'python3.9',
                        package: {
                            exclude: [],
                            include: [],
                            patterns: []
                        }
                    },
                    func2: {
                        handler: 'node-fn',
                        runtime: 'nodejs16',
                        package: {
                            exclude: [],
                            include: [],
                            patterns: []
                        }
                    }
                },
                package: {
                    exclude: [],
                    include: [],
                    patterns: []
                },
                layers: {},
                getAllLayers: jest.fn(),
                getAllFunctions: jest.fn()
            },
            pluginManager: {
                spawn: jest.fn()
            }
        };
        const plugin = new TypeScriptPlugin(slsInstance, {});
        expect(Object.keys(plugin.functions)).toEqual([
            'func2'
        ]);
    });
});
