// serverless-plugin-typescript/dist/tests/typescript.extractFileName.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const typescript_1 = require("../src/typescript");
const path = require("path");
const functions = {
    hello: {
        handler: 'tests/assets/hello.handler',
        runtime: 'nodejs10.1',
        package: {
            include: [],
            exclude: [],
            patterns: []
        }
    },
    world: {
        handler: 'tests/assets/world.handler',
        runtime: 'nodejs10.1',
        package: {
            include: [],
            exclude: [],
            patterns: []
        }
    },
    js: {
        handler: 'tests/assets/jsfile.create',
        runtime: 'nodejs10.1',
        package: {
            include: [],
            exclude: [],
            patterns: []
        }
    },
};
describe('extractFileName', () => {
    it('get function filenames from serverless service for a non-google provider', () => {
        expect((0, typescript_1.extractFileNames)(process.cwd(), 'aws', functions)).toEqual([
            'tests/assets/hello.ts',
            'tests/assets/world.ts',
            'tests/assets/jsfile.js',
        ]);
    });
    it('get function filename from serverless service for a google provider', () => {
        expect((0, typescript_1.extractFileNames)(path.join(process.cwd(), 'example'), 'google')).toEqual([
            'handler.ts'
        ]);
    });
});
